atlasdb-config/src/main/java/com/palantir/atlasdb/factory/AtlasDbRuntimeConfigRefreshable.java

```diff
@@ -0,0 +1,110 @@
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.factory;
+
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.config.AtlasDbRuntimeConfig;
+import com.palantir.common.concurrent.PTExecutors;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import com.palantir.refreshable.DefaultRefreshable;
+import com.palantir.refreshable.Refreshable;
+
+final class AtlasDbRuntimeConfigRefreshable implements AutoCloseable {
+
+    private static final Logger log = LoggerFactory.getLogger(AtlasDbRuntimeConfigRefreshable.class);
+    private static final Duration REFRESH_INTERVAL = Duration.ofSeconds(1);
+    private static final Duration GRACEFUL_SHUTDOWN = Duration.ofSeconds(5);
+    private static final AtlasDbRuntimeConfig DEFAULT_RUNTIME = AtlasDbRuntimeConfig.defaultRuntimeConfig();
+
+    private final Refreshable<AtlasDbRuntimeConfig> delegate;
+    private final Runnable closer;
+
+    private AtlasDbRuntimeConfigRefreshable(
+            Refreshable<Optional<AtlasDbRuntimeConfig>> delegate,
+            Runnable closer) {
+        this.delegate = delegate.map(config -> config.orElse(DEFAULT_RUNTIME));
+        this.closer = closer;
+    }
+
+    public Refreshable<AtlasDbRuntimeConfig> config() {
+        return delegate;
+    }
+
+    @Override
+    public void close() {
+        closer.run();
+    }
+
+    static AtlasDbRuntimeConfigRefreshable create(TransactionManagers builder) {
+        return builder.runtimeConfig()
+                .map(AtlasDbRuntimeConfigRefreshable::wrap)
+                .orElseGet(() -> {
+                    Supplier<Optional<AtlasDbRuntimeConfig>> runtimeConfig = builder.runtimeConfigSupplier()
+                            .orElse(Optional::empty);
+                    return AtlasDbRuntimeConfigRefreshable.createPolling(runtimeConfig);
+                });
+    }
+
+    private static AtlasDbRuntimeConfigRefreshable wrap(Refreshable<Optional<AtlasDbRuntimeConfig>> delegate) {
+        return new AtlasDbRuntimeConfigRefreshable(delegate, () -> {
+        });
+    }
+
+    @SuppressWarnings("FutureReturnValueIgnored")
+    private static AtlasDbRuntimeConfigRefreshable createPolling(Supplier<Optional<AtlasDbRuntimeConfig>> config) {
+        DefaultRefreshable<Optional<AtlasDbRuntimeConfig>> refreshable = new DefaultRefreshable<>(call(config));
+
+        ScheduledExecutorService executor = PTExecutors.newSingleThreadScheduledExecutor();
+        executor.scheduleWithFixedDelay(
+                () -> {
+                    try {
+                        refreshable.update(config.get());
+                    } catch (Throwable e) {
+                        // This should not occur in practice
```

Comment:
Famous last words: the danger here is that we write some sort of weird update code (since DefaultRefreshable.update propagates to all the children) and this throws, and we won't find out about it. This is a bit of a change in behavior, but I don't feel strongly that this is such high risk.

Reply:
Not sure I follow: do we mean a subscriber? If a subscriber throws it does look like the world could be in an inconsistent state, but right now I think we're fine

Reply:
Yes, I mean that Refreshables remember everyone who called #map on them and propagate the updates. Yes you could be in an inconsistent state, but I think this is by design of refreshables.